subnet/license.go

```diff
@@ -170,6 +172,36 @@ func (lv *LicenseValidator) ValidateLicense() (*licverifier.LicenseInfo, error)
 	return lv.ParseLicense(string(licData))
 }
 
+func getDurationForNextLicenseCheck(li *licverifier.LicenseInfo) time.Duration {
+	if li.ExpiresAt.Before(time.Now()) {
+		// expired, within grace period. schedule daily
+		return time.Hour * 24
+	}
+	// not expired, schedule to check just after expiry
+	return time.Until(li.ExpiresAt.Add(time.Second))
+}
+
+func (lv *LicenseValidator) scheduleNextLicenseCheck(li *licverifier.LicenseInfo, acceptedPlans []string) error {
+	duration := getDurationForNextLicenseCheck(li)
+	timer := time.NewTimer(duration)
+	defer timer.Stop()
+
+	ctxt := context.Background()
+	for {
+		select {
+		case <-timer.C:
+			li, err := lv.ValidateEnterpriseLicense(acceptedPlans)
+			if err != nil {
+				fmt.Println("license validation failed:", err.Error())
```

Comment:
We need not print here. Application that uses this library, might not be expecting this sort of printing in the library

Reply:
Thanks, done